### PR TITLE
demo: Added mDNS services broadcasting for CoAP and Http

### PR DESCRIPTION
**Pull request by FirisSA**
Hi,

Leshan could announce its services via mDNS broadcasting for automatic discovery, especially on IPv6 networks.

Use-case :
Each leshan clients can automatically discover leshans instances on the same local network.

### CODE REVIEW

File: leshan-server-demo/pom.xml [hunk 1]
```diff
@@ -26,7 +26,21 @@ Contributors:
...
+            <url>https://jitpack.io</url>
+        </repository>
+    </repositories>
+
     <dependencies>
+
+        <dependency>
+            <groupId>com.github.jmdns</groupId>
+            <artifactId>jmdns</artifactId>
+            <version>jmdns-3.5.1</version>
```
Comment (sbernard31) [inline leshan-server-demo/pom.xml]: Why not using the maven [central one](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.jmdns%22) ?

**Comment by sbernard31** [comment]
I checked for IP issue. The version [3.4.1](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=10006) are approved by eclipse.
If needed I can ask for the 3.5.1 version or we just use the 3.4.1 version.
So step 1) is ok.

**Comment by FirisSA** [comment]
I've changed pom.xml to fetch jmdns dependency from maven central

**Comment by sbernard31** [comment]
Ok sounds good, I need to open IP issue for jmdns 3.5.1.
You need to accept [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php) and ideally signed-off your commit. (see [contributon guide](https://github.com/eclipse/leshan/blob/master/CONTRIBUTING.md#legal-stuff) for more details)

Sry about all this paperworks ...

**Comment by FirisSA** [comment]
Ok, great !
I've signed the ECA, so a GPG signed commit will suffice ?

**Comment by sbernard31** [comment]
No need to GPG signed commit, the "Signed-off-by:" is enough.

If you could squash all your commits in one, then rebase it on master this will be perfect :)

**Comment by sbernard31** [comment]
I create the CQ : https://dev.eclipse.org/ipzilla/show_bug.cgi?id=13821

**Comment by sbernard31** [comment]
The CQ was approved.
So I integrated this PR in master (squashed commit : 8c653036fcb43793ec3fdf68a2cc22d89665745d)

**Pull request closed without merging**